save-backend/src/main/kotlin/com/saveourtool/save/backend/service/vulnerability/VulnerabilityService.kt

```diff
@@ -297,10 +302,11 @@ class VulnerabilityService(
     /**
      * @param name name of project
      * @param vulnerabilityName name of vulnerability
-     * @return Unit
+     * @return updated [Vulnerability]
      */
     @Transactional
     fun deleteProject(name: String, vulnerabilityName: String) = vulnerabilityProjectRepository.deleteByNameAndVulnerabilityName(name, vulnerabilityName)
+        .let { vulnerabilityRepository.findByName(name) }
```

Comment:
Here method vulnerabilityRepository.findByName(name)
will be executed before the delete method. It's better to get a vulnerability in the controller itself

Reply:
Why will it?

Reply:
Removed this line.

It might be nice it you continue these in PR with `save` button removal

Reply:
The data will only be committed after the transaction is closed

Reply:
Oh, got it, thanks

Reply:
> The data will only be committed in db after the transaction is closed

yeap, but we select inside the same transaction -- it should take into account changes from this transaction

Reply:
> > The data will only be committed in db after the transaction is closed
> 
> yeap, but we select inside the same transaction -- it should take into account changes from this transaction

Yes, but in order for these changes to be visible in the transaction itself, you must call the forced  SaveAndFlush or DeleteAndFlush methods

Reply:
looks like you re right in general, but it should be visible by default is due to `auto flush` according to https://stackoverflow.com/a/56816364.

so yes, better to use `saveAndFlush` in such cases